llvm/utils/TableGen/SearchableTableEmitter.cpp

```diff
@@ -793,7 +828,8 @@ void SearchableTableEmitter::run(raw_ostream &OS) {
     Table.Indices.push_back(
         parseSearchIndex(Table, IndexRec->getValue("Key"), IndexRec->getName(),
                          IndexRec->getValueAsListOfStrings("Key"),
-                         IndexRec->getValueAsBit("EarlyOut")));
+                         IndexRec->getValueAsBit("EarlyOut"),
+                         IndexRec->getValueAsBit("ReturnRange")));
```

Comment:
Can we just pass `false` here instead of calling `IndexRec->getValueAsBit("ReturnRange")`? Would that remove the need for having `ReturnRange` in the `SearchIndex` class in tablegen?

Reply:
This would still require to have `ReturnRange` field in `struct SearchIndex` defined in the SearchableTableEmitter file. I have made the changes. Do let me know if this will be fine

Reply:
I do not have commit access yet, if changes are good, can you merge this PR and PR#97287 on my behalf?

Reply:
> This would still require to have `ReturnRange` field in `struct SearchIndex` defined in the SearchableTableEmitter file. I have made the changes. Do let me know if this will be fine

That's fine. I just didn't want to expose something in the .td file that would always error if we could avoid it. The class in SearchTableEmitter isn't as user facing as the .td file.